libarchive/archive_write_disk_posix.c

```diff
@@ -406,6 +408,30 @@ static ssize_t	_archive_write_disk_data(struct archive *, const void *,
 static ssize_t	_archive_write_disk_data_block(struct archive *, const void *,
 		    size_t, int64_t);
 
+static int
+la_mktemp(struct archive_write_disk *a)
+{
+	int oerrno, fd;
+	mode_t mode;
+
+	archive_string_empty(&a->_tmpname_data);
+	archive_string_sprintf(&a->_tmpname_data, "%sXXXXXX", a->name);
```

Comment:
Did you consider adding a dot at the start of the file name to by hidden by default?
Rsync uses a template ".%s.XXXXXX"
In NetBSD pax you use the template "%s.XXXXXX" which is contains a dot after the file name which you probably have missed.

Reply:
I am happy to do it either way. The pro of adding a leading . is that it is invisible to casual listing while updating; the cons is that invisible files can accumulate and linger. Which do you prefer? Leading, middle, or no .?